### ISSUE TITLE
Build files environment variable or option

### ISSUE THREAD

**Issue by RaitoBezarius** [nuxt/nuxt#20714]
### Describe the feature

Hello there, I am a NixOS (https://nixos.org/) developer and we package sometimes software that uses Nuxt as a server runtime, unfortunately, because we enforce scoping and many things, we cannot guarantee that `.nuxt` is in the same current working directory, it is often locked in a immutable folder.

Therefore, we are always stuck whenever we want to run `nuxt start`, could there be a Nuxt environment variable to redirect the build files somewhere else? Thank you!

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by danielroe** [comment]
Could you give an example? Is are these Nuxt 3 or Nuxt 2 packages - or both? How much control do you have of build process - could you create a file in the working directory, such as a `.nuxtrc` file?

**Comment by RaitoBezarius** [comment]
> Could you give an example? Is are these Nuxt 3 or Nuxt 2 packages - or both? How much control do you have of build process - could you create a file in the working directory, such as a `.nuxtrc` file?

From what I see, it's a Nuxt 2 project.

I can create a `.nuxtrc` in the working directory from where I run the systemd unit or I can create it in the package bundle containing the `.nuxt` folder which becomes immutable.

The only thing I don't really have control over is: when I lock the package as an immutable package, I cannot write to it anymore.

The only way to workaround the current state of things is to do a bunch of symbolic links, which is not very satisfying from a bundling point of view.

**Comment by danielroe** [comment]
@RaitoBezarius If you can give me some code I can run, or a release process to point me to, I can try to give you a better answer.

But in short, you can override the project `buildDir` by adding a `.nuxtrc` in the user folder (or root folder of nuxt project before building) with the following content:

```
buildDir=/path/to/my/build/dir
```

**Comment by RaitoBezarius** [comment]
> @RaitoBezarius If you can give me some code I can run, or a release process to point me to, I can try to give you a better answer.
> 
> But in short, you can override the project `buildDir` by adding a `.nuxtrc` in the user folder (or root folder of nuxt project before building) with the following content:
> 
> ```
> buildDir=/path/to/my/build/dir
> ```

Well, we perform our build like this: https://github.com/NixOS/nixpkgs/pull/215865/files#diff-fce5f1f118cad60e0770d9849f978d600a93cf37a483da0a5ca8bd11378366c3R38-R77 (but it's quite complicated due to Baserow being complicated and it requires Nix knowledge).

But when we run it: https://github.com/NixOS/nixpkgs/pull/215865/files#diff-5e9c0be333088c85a5163a95eb72db1812e318552796593c09049bd4c28b6f35R147-R171

My biggest problem you can see I have to `ln` the `.nuxt` so that the cwd can find it. Also, I don't know if `nuxt` expect its cwd to be writable, if not, I could change cwd into the package with no problem, but it cannot write anything in the cwd (again, it's immutable, `+i` xattrs): https://github.com/NixOS/nixpkgs/pull/215865/files#diff-5e9c0be333088c85a5163a95eb72db1812e318552796593c09049bd4c28b6f35R148

Changing the `buildDir` might not solve the underlying issue because I want this to be part of the package and lock it.

Maybe my XY problem (apologies) is: can I force nuxt to **not write** anything in the cwd? If so, this is all solved.

**Comment by danielroe** [comment]
Yes, you can set a `buildDir` which is outside the `cwd`, and Nuxt should not manipulate the `cwd` at all.

You will need to update your `tsconfig.json` to extend the generated `.nuxt/tsconfig.json` so that Vue/Vite can resolve the types.

**Comment by hacknug** [comment]
@danielroe you provided a solution, it has no more comments, and it is about v2. can close?

**Comment by danielroe** [comment]
nuxtathon closed @hacknug
